feat: add access control to Lemon UI components

zlwaterfield opened the pull request:
Introduces access control props (userAccessLevel, minAccessLevel, resourceType) to LemonButton, LemonSelect, and LemonSwitch components. Components now automatically disable themselves and show a tooltip if the user's access level does not meet the required minimum for the specified resource type. Storybook stories have been updated to demonstrate the new access control functionality.

We currently use AccessControlledLemonButton but it's messy in the code and added the props directly to the underlying components like this makes it much cleaner.

zlwaterfield committed:
Add access control to Lemon UI components

Introduces access control props (userAccessLevel, minAccessLevel, resourceType) to LemonButton, LemonSelect, and LemonSwitch components. Components now automatically disable themselves and show a tooltip if the user's access level does not meet the required minimum for the specified resource type. Storybook stories have been updated to demonstrate the new access control functionality.

zlwaterfield committed:
Refactor access control props to use AccessControlLevel type

Updated LemonButton, LemonSelect, and LemonSwitch components to use the AccessControlLevel type for userAccessLevel and minAccessLevel props instead of string literals. This improves type safety and consistency across access control implementations.

zlwaterfield committed:
Update LemonButton.stories.tsx

github-actions[bot] committed:
Update UI snapshots for `chromium` (9)

github-actions[bot] committed:
Update UI snapshots for `chromium` (14)

github-actions[bot] committed:
Update UI snapshots for `chromium` (6)

zlwaterfield committed:
Update LemonSwitch.stories.tsx

zlwaterfield committed:
Merge branch 'master' into zach/access-controlled-components

adamleithp reviewed (approved):
I do agree with greptile's nit of not "mutating disabled prop`, but all good for now

zlwaterfield committed:
Refactor access control props in Lemon UI components

Replaces individual access control props (userAccessLevel, minAccessLevel, resourceType) with a single 'accessControl' object in LemonButton, LemonSelect, and LemonSwitch components and their stories. This improves prop clarity and simplifies permission checking logic.

zlwaterfield committed:
Import AccessControlLevel in LemonButton component

zlwaterfield committed:
Refactor imports in LemonSelect.tsx

zlwaterfield committed:
Import AccessControlLevel in LemonSwitch component

yasen-posthog commented on frontend/src/lib/lemon-ui/LemonSelect/LemonSelect.tsx:
```diff
@@ -81,6 +85,10 @@ export interface LemonSelectPropsBase<T>
     menu?: Pick<LemonMenuProps, 'className' | 'closeParentPopoverOnClickInside'>
     visible?: LemonDropdownProps['visible']
     startVisible?: LemonDropdownProps['startVisible']
+    /** Access control props for automatic permission checking */
```
Hehe, I meant the comment, but Alex's suggestion is spot on I think 👌

zlwaterfield committed:
Merge branch 'master' into zach/access-controlled-components

zlwaterfield committed:
fix ts issues

zlwaterfield committed:
LINT
